autoemulate/emulators/neural_networks/mlp.py

```diff
@@ -28,30 +30,25 @@ def __init__(
         modules = []
         for hidden_size in hidden_sizes:
             modules.append(nn.Linear(in_features=input_size, out_features=hidden_size))
-            modules.append(nn.ReLU())
+            modules.append(hidden_activation())
             input_size = hidden_size
         modules.append(nn.Linear(in_features=input_size, out_features=output_size))
         self.model = nn.Sequential(*modules)
 
     def get_grid_params(self, search_type: str = "random"):
+        param_space = {
+            "lr": loguniform(1e-06, 0.01),
+            "max_epochs": np.arange(10, 110, 10).tolist(),
+            "batch_size": np.arange(2, 128, 2).tolist(),
+            "module__hidden_sizes": [(50,), (100,), (100, 100), (100, 200), (200, 200)],
+            "module__hidden_activation": [nn.ReLU, nn.Tanh, nn.Sigmoid, nn.GELU],
+            "optimizer__weight_decay": (1 / 10 ** np.arange(1, 10)).tolist(),
+        }
```

Comment:
This currently fails for the [bayes search](https://scikit-optimize.github.io/stable/modules/generated/skopt.BayesSearchCV.html) (see below), which needs a slightly different `param_space` definition (see other models). It's a bit annoying but we have essentially define the `search_space` twice for now.

```
em = AutoEmulate()
em.setup(X, y, param_search_type="bayes", model_subset=["NeuralNetTorch"], param_search=True)
em.compare()
```

Reply:
fixed the issue. also separated `module__hidden_size` and `module__hidden_layers` so that we can use `bayes` for it as well.